sdk/core/Azure.Core/api/Azure.Core.net461.cs

```diff
@@ -1201,6 +1201,12 @@ public abstract partial class ObjectSerializer
         public abstract System.Threading.Tasks.ValueTask SerializeAsync(System.IO.Stream stream, object? value, System.Type inputType, System.Threading.CancellationToken cancellationToken);
         public virtual System.Threading.Tasks.ValueTask<System.BinaryData> SerializeAsync(object? value, System.Type? inputType = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
     }
+    [System.AttributeUsageAttribute(System.AttributeTargets.Class)]
+    public sealed partial class UnknownSubclassAttribute : System.Attribute
```

Comment:
It feels weird to name this "UknownSubclass" when the subclass is actually known. That's what we are pointing to. So I guess I don't get what "Unknown" means in this context.

(Note: I'm just a bystander here, so take my feedback as just a random opinion.)

Reply:
This is for polymorphic forward compat.  In v1 we have _Animal := Cat | Dog_ which turns into `abstract class Animal`, `class Cat : Animal`, `class Dog : Animal`.  In v2 they add `class Fish : Animal`.  If someone creates a bunch of pets in v2 of the service, including fish, and our app using v1 of the service calls `client.GetAllPets();` what do we return for those fish?  We can't instantiate `Animal` because it's abstract.  Instead we have a convention where we create an `internal class UnknownAnimal : Animal` that we can instantiate and return from v1 for resource kinds that never existed when v1 was built.  That's what this attribute is intended to signal to our general purpose serializers in an AOT friendly way.  We of course debated making `Animal` not abstract as well, but we really like the indicator to customers they need to write forward compat friendly code (i.e., have a reasonable default/fall through case).

It's fair that `UnknownSubclass` / `UnknownAnimal` still might not be a super clear indicator a time traveler just screwed with your ~~global variables~~ cloud resources.  I'm not sure that `FutureSubclass`, `ForwardCompatSubclass`, `UnknownResourceClass`, etc., are necessarily better.  Most customers won't care about the attribute itself though, other than seeing it's a thing in docs.

The `internal class UnknownFoo`s are really where we should invest (i.e., a DebuggerDisplay or maybe ToString that clearly explains the discriminator wasn't known to the library as of `Max(ServiceVersion)`).  @m-nash - We should also make sure the ref docs for `UnknownSubclass` have a helpful explanation for anyone who clicks on it from a random base class page since the attribute is public.